packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetryEndToEnd.spec.realsvc.ts

```diff
@@ -0,0 +1,241 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TinyliciousClient } from "@fluidframework/tinylicious-client/internal";
+import { SharedTree } from "@fluid-experimental/tree";
+import type Sinon from "sinon";
+import { spy } from "sinon";
+import { expect } from "chai";
+import { startTelemetry } from "../factory/index.js";
+import {
+	ContainerTelemetryEventNames,
+	type ContainerConnectedTelemetry,
+	type ContainerDisconnectedTelemetry,
+	type ContainerDisposedTelemetry,
+	type IFluidTelemetry,
+	type ITelemetryConsumer,
+} from "../index.js";
+
+// This test suite creates an actual IFluidContainer and confirms events are fired with the expected names during the expected events
+
+describe("container telemetry E2E", () => {
+	let schema;
+	let tinyliciousClient: TinyliciousClient;
+	let telemetryConsumerConsumeSpy: Sinon.SinonSpy;
+	let testTelemetryConsumer: ITelemetryConsumer;
+
+	// Simple test class that will be used as the telemetry consumer.
+	class TestTelemetryConsumer implements ITelemetryConsumer {
+		/**
+		 * Takes the incoming {@link IFluidTelemetry} and sends it to Azure App Insights
+		 */
+		public consume(event: IFluidTelemetry): void {
+			return;
+		}
+	}
+
+	before(() => {
+		tinyliciousClient = new TinyliciousClient({ connection: { port: 7070 } });
+		schema = {
+			initialObjects: {
+				sharedMap1: SharedTree,
+			},
+		};
+
+		testTelemetryConsumer = new TestTelemetryConsumer();
+		telemetryConsumerConsumeSpy = spy(testTelemetryConsumer, "consume");
+	});
+
+	beforeEach(() => {
+		// We need to reset the telemetry consumer spy for each test so the trackEvent calls
+		// from the last test don't bleed into the next one.
+		telemetryConsumerConsumeSpy.resetHistory();
+	});
+
+	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry using AppInsightsTelemetryConsumer", async () => {
```

Comment:
Similar for the other tests.
```suggestion
	it("IFluid container's 'connected' system event produces expected ContainerConnectedTelemetry", async () => {
```